mbuild/compound.py

```diff
@@ -1041,6 +1041,66 @@ def center(self):
         if np.all(np.isfinite(self.xyz)):
             return np.mean(self.xyz, axis=0)
 
+    @property
+    def center_of_mass(self):
+        """Returns the center of mass of a Compound in cartesian coordinates.
+
+        Returns
+        -------
+        np.ndarray, shape=(3,), dtype=float
+           The cartesian center of the Compound based on its Particles and molecular weight.
+            """
+        guessed_elements = set()
+        mass_vector = np.array([0., 0., 0.])
+        total_mass = 0
+        for atom in self.particles():
+            name = ''.join(char for char in atom.name if not char.isdigit())
+            try:
+                AtomicNum[atom.name]
+            except KeyError:
+                element = element_by_name(atom.name)
```

Comment:
Standard convention for mBuild (which aids in the interworking with Foyer) is to use atomic symbols for `Particle` names and to preface names with an underscore for non-atomistic particles. So I'm thinking we don't want to do any guessing of element names (although stripping digits is fine I suppose).

If any `atom.name` returns a `KeyError` when attempting `AtomicNum[atom.name]` then the user either isn't following standard mBuild convention or is operating on a non-atomistic system. In this case, I think we should just go ahead and return `self.center` along with a warning message of something like `'Non-element {} detected in Compound. Returning Compound.center().'.format(atom.name)`